src/txmempool.h

```diff
@@ -62,6 +68,27 @@ class CTxMemPoolEntry
     size_t DynamicMemoryUsage() const { return nUsageSize; }
 };
 
+// extracts a TxMemPoolEntry's transaction hash
+struct mempoolentry_txid
+{
+    typedef uint256 result_type;
+    result_type operator() (const CTxMemPoolEntry &entry) const
+    {
+        return entry.GetTx().GetHash();
+    }
+};
+
+class CompareTxMemPoolEntryByFee
```

Comment:
Actually it does use the time too: https://github.com/bitcoin/bitcoin/pull/6421/files#diff-8304b3e94624036c3673f31eeb7e9de0R88